src/frontends/tensorflow_common/src/op/binary_op.cpp

```diff
@@ -142,6 +142,30 @@ OutputVector translate_mul_op(const NodeContext& node) {
     set_node_name(node.get_name(), result);
     return {result};
 }
+
+OutputVector translate_sub_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"Sub"}, true);
+    auto lhs = node.get_input(0);
+    auto rhs = node.get_input(1);
+
+    auto result = make_shared<v1::Subtract>(lhs, rhs);
+
+    auto complex_type_mark_lhs = as_type_ptr<ComplexTypeMark>(lhs.get_node_shared_ptr());
+    auto complex_type_mark_rhs = as_type_ptr<ComplexTypeMark>(rhs.get_node_shared_ptr());
+
+    if (complex_type_mark_lhs || complex_type_mark_rhs) {
+        auto complex_result =
+            make_shared<ComplexTypeMark>(result,
+                                         complex_type_mark_lhs ? complex_type_mark_lhs->get_complex_part_type()
+                                                               : complex_type_mark_rhs->get_complex_part_type());
+        set_node_name(node.get_name(), complex_result);
+        return {complex_result};
+    }
```

Comment:
please check if your code works for complex type using layer test. I think - no because `ComplexTypeMark` is not properly handled

Reply:
@rkazants 

I tried running the layer_test,

``` class TestComplexSub(CommonTFLayerTest):
    def _prepare_input(self, inputs_info):

        rng = np.random.default_rng()
        assert "param_real_1:0" in inputs_info
        assert "param_imag_1:0" in inputs_info
        assert "param_real_2:0" in inputs_info
        assert "param_imag_2:0" in inputs_info
        param_real_shape_1 = inputs_info["param_real_1:0"]
        param_imag_shape_1 = inputs_info["param_imag_1:0"]
        param_real_shape_2 = inputs_info["param_real_2:0"]
        param_imag_shape_2 = inputs_info["param_imag_2:0"]

        inputs_data = {}
        inputs_data["param_real_1:0"] = (
            4 * rng.random(param_real_shape_1).astype(np.float32) - 2
        )
        inputs_data["param_imag_1:0"] = (
            4 * rng.random(param_imag_shape_1).astype(np.float32) - 2
        )
        inputs_data["param_real_2:0"] = (
            4 * rng.random(param_real_shape_2).astype(np.float32) - 2
        )
        inputs_data["param_imag_2:0"] = (
            4 * rng.random(param_imag_shape_2).astype(np.float32) - 2
        )

        return inputs_data

    def create_complex_sub_net(self, input_shape):
        import tensorflow as tf

        tf.compat.v1.reset_default_graph()
        # Create the graph and model
        with tf.compat.v1.Session() as sess:
            param_real1 = tf.compat.v1.placeholder(
                np.float32, input_shape, "param_real_1"
            )
            param_imag1 = tf.compat.v1.placeholder(
                np.float32, input_shape, "param_imag_1"
            )
            param_real2 = tf.compat.v1.placeholder(
                np.float32, input_shape, "param_real_2"
            )
            param_imag2 = tf.compat.v1.placeholder(
                np.float32, input_shape, "param_imag_2"
            )
            complex1 = tf.raw_ops.Complex(real=param_real1, imag=param_imag1)
            complex2 = tf.raw_ops.Complex(real=param_real2, imag=param_imag2)

            sub = tf.raw_ops.Sub(x=complex1, y=complex2, name="complex_sub")
            real = tf.raw_ops.Real(input=sub)
            imag = tf.raw_ops.Imag(input=sub)

            tf.compat.v1.global_variables_initializer()
            tf_net = sess.graph_def
        return tf_net, None

    test_data_basic = [
        dict(input_shape=[]),
        dict(input_shape=[2]),
        dict(input_shape=[1, 3]),
        dict(input_shape=[2, 3, 4]),
        dict(input_shape=[3, 4, 5, 6]),
    ]

    @pytest.mark.parametrize("params", test_data_basic)
    @pytest.mark.precommit_tf_fe
    @pytest.mark.nightly
    def test_complex_sub(
        self, params, ie_device, precision, ir_version, temp_dir, use_legacy_frontend
    ):
        self._test(
            *self.create_complex_sub_net(**params),
            ie_device,
            precision,
            ir_version,
            temp_dir=temp_dir,
            use_legacy_frontend=use_legacy_frontend,
        )
```




 but I get the error ->

```
E       openvino._pyopenvino.OpConversionFailure: Check 'is_conversion_successful' failed at src/frontends/tensorflow/src/frontend.cpp:428:
E       FrontEnd API failed with OpConversionFailure:
E       [TensorFlow Frontend] Internal error, conversion is failed for Real operation with a message:
E       Check 'complex_type_mark' failed at src/frontends/tensorflow_common/src/op/real_imag.cpp:36:
E       FrontEnd API failed with OpValidationFailureWhile validating node 'Real':
E       [TensorFlow Frontend] internal error: ComplexTypeMark is not set at the input of Real
E       
E       [TensorFlow Frontend] Internal error, conversion is failed for Imag operation with a message:
E       Check 'complex_type_mark' failed at src/frontends/tensorflow_common/src/op/real_imag.cpp:36:
E       FrontEnd API failed with OpValidationFailureWhile validating node 'Imag':
E       [TensorFlow Frontend] internal error: ComplexTypeMark is not set at the input of Imag
E       
E       [TensorFlow Frontend] Internal error, conversion is failed for Sub operation with a message:
E       Check '!has_input_complex_type || supported_complex' failed at src/frontends/tensorflow_common/src/utils.cpp:285:
E       FrontEnd API failed with OpValidationFailureWhile validating node 'Sub':
E       [TensorFlow Frontend] internal error: translator for Sub does not support input complex type
E       
E       [TensorFlow Frontend] Internal error, no translator found for operation(s): ComplexTypeMark
E       To facilitate the conversion of unsupported operations, refer to Frontend Extension documentation: https://docs.openvino.ai/latest/openvino_docs_Extensibility_UG_Frontend_Extensions.html

/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/openvino/frontend/frontend.py:18: OpConversionFailure
```

I've seen this issue mentioned in other threads too, but no solution. 
Can you give me any pointers on how to resolve this

Reply:
`ComplexType` is just a marker that is not supported for inference. It is like a splitter of a graph into two parts: before `ComplexTypeMark` complex tensors are represented as floating-point type tensor; after this mark they are not yet handled.
So you should work with `complex_type_mark_lhs->input_value(0)` that provides this floating-point tensor and pass it to new `Subtract` node. The same for `rhs` node. This way you removed this marker before `Subtract` node but you should insert this marker after the subtract result because it will be handled by next operation translators.

Reply:
@rkazants 

I've updated the code based on your comment where I take the FP tensors, subtract and then set ComplexTypeMark on the result, but I still get the layer_tests errors.

Modified code->
```
OutputVector translate_sub_op(const NodeContext& node) {
    default_op_checks(node, 2, {"Sub"}, true);
    auto lhs = node.get_input(0);
    auto rhs = node.get_input(1);

    auto complex_type_mark_lhs = as_type_ptr<ComplexTypeMark>(lhs.get_node_shared_ptr());
    auto complex_type_mark_rhs = as_type_ptr<ComplexTypeMark>(rhs.get_node_shared_ptr());

    if (complex_type_mark_lhs || complex_type_mark_rhs) {
        lhs = complex_type_mark_lhs->input_value(0);
        rhs = complex_type_mark_rhs->input_value(0);

        element::Type complex_part_type = complex_type_mark_lhs ? complex_type_mark_lhs->get_complex_part_type()
                                                                : complex_type_mark_rhs->get_complex_part_type();

        auto result = make_shared<v1::Subtract>(lhs, rhs);
        set_node_name(node.get_name(), result);

        auto complex_result = make_shared<ComplexTypeMark>(result->output(0), complex_part_type);
        return {complex_result};
    }
    auto result = make_shared<v1::Subtract>(lhs, rhs);
    set_node_name(node.get_name(), result);
    return {result};
}
```